### PR TITLE
Change `error` from directive to keyword.

### LINKED ISSUE #11214: Make `error` as a keyword in 0.9.0
The following test case shouldn't parse / compile.

```solidity
contract C {
	error error(uint error);
	function f() public {
		revert error(5);
	}
}
```

### PR DESCRIPTION
**Pull request by christianparpart**
Closes #11214.

Makes `error` a keyword, prohibiting constructs like mentioned in the above ticket.

## Checklist

- [x] Changelog entry

### CODE REVIEW

**Comment by chriseth** [comment]
This is a breaking change.

File: docs/grammar/SolidityLexer.g4 [hunk 1]
```diff
@@ -29,7 +29,7 @@ Do: 'do';
 Else: 'else';
 Emit: 'emit';
 Enum: 'enum';
-Error: 'error'; // not a real keyword
+Error: 'error';
 Revert: 'revert'; // not a real keyword
```
Comment (chriseth) [inline docs/grammar/SolidityLexer.g4:33]: Should we also make revert a keyword? Then we lose the revert function and it is impossible to have code that can be compiled with 0.8.0 and 0.9.0.

Comment (christianparpart) [reply docs/grammar/SolidityLexer.g4:33]: @chriseth I think revert should stay as a regular symbol -for now-, as it is used like a function. OTOH i think it would make sense to make the language less prone to accidental errors (people misunderstanding code when they see the name `revert(...)`). But that could then be done in a later PR or even the next major release after that.

Comment (cameel) [reply docs/grammar/SolidityLexer.g4:33]: For me the question would be whether we want to remove the `revert()` function. If we do, then I'd make it a keyword. If we don't then I guess it's not an option anyway.

File: test/libsolidity/semanticTests/reverts/error_struct.sol [hunk 2]
```diff
@@ -1,17 +1,10 @@
-struct error { uint error; }
 contract C {
 	error test();
-	error _struct;
```
Comment (cameel) [inline test/libsolidity/semanticTests/reverts/error_struct.sol:4]: The fact that a test case named `error_struct.sol` no longer contains an `error` struct makes it somewhat pointless ;)

File: test/libsolidity/semanticTests/errors/weird_name.sol [hunk 3]
```diff
@@ -1,10 +0,0 @@
-error error(uint a);
-contract C {
```
Comment (cameel) [inline test/libsolidity/semanticTests/errors/weird_name.sol:2]: Could you keep this case as a syntax test?

**Comment by cameel** [comment]
Needs rebase. It's based on a commit where we had problems with the macOS build in `develop`.

File: test/libsolidity/semanticTests/reverts/error_struct.sol (hunk 2 above)
Comment (hrkrshnn) [reply test/libsolidity/semanticTests/reverts/error_struct.sol:4]: Agree, this could be removed.

**Review by hrkrshnn — COMMENTED** [review]
Looks good. Might be useful to create a file `090-breaking-changes.rst` and add something about errors, just like in `080-breaking-changes.rst`.

File: test/libsolidity/semanticTests/reverts/error_struct.sol (hunk 2 above)
Comment (cameel) [reply test/libsolidity/semanticTests/reverts/error_struct.sol:4]: Or we could turn it into a negative syntax test that makes sure that this generates an error.

**Comment by christianparpart** [comment]
> Looks good. Might be useful to create a file `090-breaking-changes.rst` and add something about errors, just like in `080-breaking-changes.rst`.

Done. And also added the errors change into the "new restrictions" section. If everything else is okay, we could at least reduce the PR count by 1. :-)

File: test/libsolidity/semanticTests/errors/weird_name.sol (hunk 3 above)
Comment (hrkrshnn) [reply test/libsolidity/semanticTests/errors/weird_name.sol:2]: @christianparpart this is still missing. Maybe the existing syntax test is enough, i.e., `struct_named_error`.

**Review by hrkrshnn — APPROVED** [review]
A suggestion by @cameel is missing. But approving anyway.

File: test/libsolidity/semanticTests/errors/weird_name.sol (hunk 3 above)
Comment (hrkrshnn) [reply test/libsolidity/semanticTests/errors/weird_name.sol:2]: I'll add this, and push now.

File: test/libsolidity/syntaxTests/errors/weird_name.sol [hunk 4]
```diff
@@ -0,0 +1,8 @@
+error error(uint a);
+contract C {
+    function f() public pure {
+        revert error(2);
+    }
+}
+// ----
+// ParserError 2314: (6-11): Expected identifier but got 'error'
```
Comment (hrkrshnn) [inline test/libsolidity/syntaxTests/errors/weird_name.sol:8]: Added this.

**Review by hrkrshnn — APPROVED** [review]

File: Changelog.md [hunk 5]
```diff
@@ -1,15 +1,9 @@
-### 0.8.5 (unreleased)
-
-Language Features:
```
Comment (axic) [inline Changelog.md:3]: Huh, why were these removed?

File: docs/090-breaking-changes.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
+********************************
+Solidity v0.8.0 Breaking Changes
```
Comment (axic) [inline docs/090-breaking-changes.rst:2]: 0.9.0 and not 0.8.0.

File: test/libsolidity/semanticTests/reverts/error_struct.sol [hunk 7]
```diff
@@ -1,18 +0,0 @@
-struct error { uint error; }
```
Comment (axic) [inline test/libsolidity/semanticTests/reverts/error_struct.sol:1]: Why wasn't this also moved to syntax tests?

**Comment by axic** [comment]
Why was this an important change? Why do we keep those others things as non-keywords?

File: test/libsolidity/semanticTests/reverts/error_struct.sol (hunk 7 above)
Comment (hrkrshnn) [reply test/libsolidity/semanticTests/reverts/error_struct.sol:1]: We already have two tests for parser errors for `error`, and this one doesn't really add anything more, I think.

**Comment by chriseth** [comment]
@axic what other things? I'm fine reverting this change  - let's discuss next week

File: Changelog.md (hunk 5 above)
Comment (hrkrshnn) [reply Changelog.md:3]: I assumed we were taking breaking off 0.8.4, but that is obviously incorrect. @christianparpart could you please fix this?
